Block/Adminhtml/System/Config/FeePlansConfigFrontModel.php

```diff
@@ -0,0 +1,210 @@
+                'renderer' => $this->getTextStyle(),
+            ]
+        );
+        $this->addColumn(
+            'custom_min_purchase_amount',
+            [
+                'label' => __('Min display amount'),
+                'style' => self::SMALL_INPUT_COLUMN,
+                'class' => 'validate-number'
+            ]
+        );
+        $this->addColumn(
+            'custom_max_purchase_amount',
+            [
+                'label' => __('Max display amount'),
+                'style' => self::SMALL_INPUT_COLUMN,
+                'class' => 'validate-number'
+            ]
+        );
+        $this->addColumn(
+            'max_purchase_amount',
+            [
+                'label' => __('Max amount'),
+                'style' => self::SMALL_STRING_COLUMN,
+                'renderer' => $this->getTextStyle(),
+            ]
+        );
+        $this->_addAfter = false;
+        $this->_addButtonLabel = __('Add More');
+    }
+
+    /**
+     * @param DataObject $row
+     *
+     * @return void
+     * @throws LocalizedException
+     */
+    protected function _prepareArrayRow(DataObject $row): void
+    {
+        $options = [];
+        $selectFieldData = $row->getSelectField();
+        if ($selectFieldData !== null) {
+            $options['option_' . $this->getSelectFieldOptions()->calcOptionHash($selectFieldData)] = 'selected="selected"';
+        }
+        $row->setData('option_extra_attrs', $options);
+    }
+
+
+    /**
+     * @return BlockInterface
+     * @throws LocalizedException
+     */
+    private function getSelectFieldOptions(): BlockInterface
+    {
+        if (!$this->selectOptions) {
+            $this->selectOptions = $this->getLayout()->createBlock(
+                DynamicRowEnableSelect::class,
+                '',
+                ['data' => []]
+            );
+        }
+        return $this->selectOptions;
+    }
+
+    /**
+     * @return BlockInterface
+     * @throws LocalizedException
+     */
+    private function getTextStyle(): BlockInterface
+    {
+        if (!$this->renderString) {
+            $this->renderString = $this->getLayout()->createBlock(
+                DynamicRowText::class,
+                '',
+                ['data' => []]
+            );
+        }
+        return $this->renderString;
+    }
+    /**
+     * Get the grid and scripts contents
+     *
+     * @param AbstractElement $element
+     *
+     * @return string
+     */
+    protected function _getElementHtml(AbstractElement $element)
+    {
+        $element->setComment($this->getHtmlComment($element->getValue()));
+        $this->setElement($element);
+        $html = $this->_toHtml();
+        $this->_arrayRowsCache = null;
+        // doh, the object is used as singleton!
+        return $html;
+    }
+
+    private function getHtmlComment($feePlans): string
+    {
+        $html = '<div style="font-size: x-small" data-toggle="collapse" ><p>' . __('Fees applied to each transaction :') . '</p>';
```

Comment:
just a comment (not required)

I prefer `sprintf()` to `.` concat

to discuss on a rainy day 😉